syntax = "proto3";

package M;

enum RpcType {
  RPC_TYPE_RPC_REQUEST_UNSPECIFIED = 0;
  RPC_TYPE_RPC_REQUEST = 1;
  RPC_TYPE_RPC_RESPONSE = 2;
}

message RpcHeader {
  optional RpcType type = 1;
  optional string func = 2;
  optional string timeout = 3;
}

message HelloReq {
  optional string name = 1;
}

message HelloRes {
  optional string name = 1;
}
